app/models/knight.rb

```diff
@@ -1,2 +1,9 @@
 class Knight < Piece
+  def valid_move?(destination_x, destination_y)
+    return false unless destination_x.between?(0, 7) && destination_y.between?(0, 7)
+    difference_x = destination_x - position_x
+    difference_y = destination_y - position_y
+    return true if difference_x.abs == 1 && difference_y.abs == 2 || difference_x.abs == 2 && difference_y.abs == 1
+    false
```

Comment:
Looks great. One simplification you could do.

Since `difference_x.abs == 1 && difference_y.abs == 2 || difference_x.abs == 2 && difference_y.abs == 1` always evaluates to `true` or `false`, it could be the last line of your method. And this code would behave the same as is does now.

😄

Reply:
I tried removing line 7 (`false`), but then three tests failed that were expecting a `false` response. For these three tests, Rspec expected `false` but got `nil`.

Reply:
Ah, never mind. I see what you mean now. At first, I was leaving `return true if` in the last line, but the statement `difference_x.abs == 1 && difference_y.abs == 2 || difference_x.abs == 2 && difference_y.abs == 1` on its own evaluates to `true` or `false`, so that can be the last line on its own, and it will return the proper value.